lib/house.rb

```diff
@@ -0,0 +1,42 @@
+class House
+
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price[1..-1].to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    if @price > 500000
+      true
+    else
+      false
+    end
+  end
+
+  def area
+    room_areas = []
+    @rooms.each do |room|
+      room_areas << room.area
+    end
+    room_areas.sum
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all do |room|
+      if room.category == category
+        room
+      end
+    end
```

Comment:
Nice job choosing the best tool for the job! `find_all` is a great enumerable here. `find_all` only takes a boolean condition though, so the way this should actually  be written is as follows:

```
def rooms_from_category(category)
   @rooms.find_all do |room|
         room.category == category 
    end
end 
```

I think in this case, you got a little bit lucky, and it works anyway, but you can get yourself into trouble, setting yourself up for false positives, by putting anything other than a conditional into a `find_all`